modules/nixos.nix

```diff
@@ -34,29 +34,127 @@ in
             default.value = [ ];
           };
 
+          home-manager = lib.options.create {
+            description = "The home-manager input to use.";
+            type = lib.types.raw;
+            default.value =
+              if inputs ? home-manager
+              then inputs.home-manager.result
+              else null;
+          };
+
+          homes = lib.options.create {
+            description = "A list of homes to activate for the system.";
+            type = lib.types.list.of lib.types.string;
+            default.value = [ ];
+          };
+
           result = lib.options.create {
             description = "The created NixOS system.";
             type = lib.types.raw;
             writable = false;
-            default.value = import "${config.pkgs.path}/nixos/lib/eval-config.nix" {
-              pkgs = config.pkgs;
-              lib = config.pkgs.lib;
-              specialArgs = config.args;
-              modules = config.modules;
+            default.value = import "${submodule.config.pkgs.path}/nixos/lib/eval-config.nix" {
+              pkgs = submodule.config.pkgs;
+              lib = submodule.config.pkgs.lib;
+              specialArgs = submodule.config.args;
+              modules = submodule.config.modules;
               modulesLocation = null;
             };
           };
         };
+
+        config.modules = let
+          system = submodule.config.pkgs.system;
+          warn = builtins.warn or builtins.trace; # builtins.warn doesn't exist on some versions of nix/lix
+          warnIf = condition: message: value: if condition then warn message value else value;
+          homeManager = submodule.config.home-manager;
+        in (lib.fp.pipe [
+          (map (homeName: let
```

Comment:
Let's make sure we prefix builtins with `builtins.`.

Reply:
ahhh, you're right, I guess I shouldn't be shadowing warn here...

Reply:
renamed to `warn'` in the typical nix convention of "just close enough to not be the same thing"

Reply:
wait, unless you meant map(?) I guess you could've sorry, github's context confused me here

Reply:
ok, done that too